File: leetcode/go/first_bad_version.go
package main

//lint:ignore U1000 Ignore unused function
func firstBadVersion(n int) int {
	left := 1
	right := n

	for left < right {
		mid := left + (right-left)/2

		if isBadVersion(mid) {
			right = mid
		} else {
			left = mid + 1
		}
	}

	return left
}

func isBadVersion(version int) bool { return true }

File: leetcode/rust/src/first_bad_version.rs
#[allow(dead_code)]
fn first_bad_version(n: i64) -> i64 {
    let mut left = 0;
    let mut right = n;

    while left < right {
        let mid = left + (right - left) / 2;

        if is_bad_version(mid) {
            right = mid
        } else {
            left = mid + 1
        }
    }

    left
}

fn is_bad_version(_v: i64) -> bool {
    true
}
